cel/pkg/reconciler/celrun/celrun.go

```diff
@@ -0,0 +1,283 @@
+			run.Status.MarkRunFailed(celv1alpha1.CELRunReasonSyntaxError.String(),
+				"CEL expression %s could not be parsed", expression.Name, iss.Err())
+			return nil
+		}
+
+		// Generate an evaluable instance of the Ast within the environment
+		prg, err := env.Program(ast)
+		if err != nil {
+			logger.Errorf("CEL expression %s could not be evaluated when reconciling Run: %v", expression.Name, err)
+			run.Status.MarkRunFailed(celv1alpha1.CELRunReasonEvaluationError.String(),
+				"CEL expression %s could not be evaluated", expression.Name, err)
+			return nil
+		}
+
+		// Evaluate the CEL expression (Ast)
+		variablesMap := getVariablesMap(CELSpec.Variables)
+		out, _, err := prg.Eval(variablesMap)
+		if err != nil {
+			logger.Errorf("CEL expression %s could not be evaluated when reconciling Run: %v", expression.Name, err)
+			run.Status.MarkRunFailed(celv1alpha1.CELRunReasonEvaluationError.String(),
+				"CEL expression %s could not be evaluated", expression.Name, err)
+			return nil
+		}
+
+		// Evaluation of CEL expression was successful
+		logger.Infof("CEL expression %s evaluated successfully when reconciling Run", expression.Name)
+		status.Results = append(status.Results, v1alpha1.RunResult{
+			Name:  expression.Name,
+			Value: fmt.Sprintf("%s", out.ConvertToType(celtypes.StringType).Value()),
+		})
+	}
+
+	// All CEL expressions were evaluated successfully
+	run.Status.Results = append(run.Status.Results, status.Results...)
+	run.Status.MarkRunSucceeded(celv1alpha1.CELRunReasonEvaluationSuccess.String(),
+		"CEL expressions were evaluated successfully")
+
+	return nil
+}
+
+func (c *Reconciler) getCEL(ctx context.Context, run *v1alpha1.Run) (*metav1.ObjectMeta, *celv1alpha1.CELSpec, error) {
+	CELMeta := metav1.ObjectMeta{}
+	CELSpec := celv1alpha1.CELSpec{}
+	if run.Spec.Ref != nil && run.Spec.Ref.Name != "" {
+		// Use the k8s client to get the CEL rather than the lister.  This avoids a timing issue where
+		// the CEL is not yet in the lister cache if it is created at nearly the same time as the Run.
+		// See https://github.com/tektoncd/pipeline/issues/2740 for discussion on this issue.
+		cs, err := c.CELClientSet.CustomV1alpha1().CELs(run.Namespace).Get(ctx, run.Spec.Ref.Name, metav1.GetOptions{})
+		if err != nil {
+			run.Status.MarkRunFailed(celv1alpha1.CELRunReasonCouldntGetCEL.String(),
+				"Error retrieving CEL for Run %s/%s: %s",
+				run.Namespace, run.Name, err)
+			return nil, nil, fmt.Errorf("error retrieving CEL for Run %s: %w", fmt.Sprintf("%s/%s", run.Namespace, run.Name), err)
+		}
+		CELMeta = cs.ObjectMeta
+		CELSpec = cs.Spec
+	} else {
+		// Run does not require name but for CEL it does
+		run.Status.MarkRunFailed(celv1alpha1.CELRunReasonCouldntGetCEL.String(),
+			"Missing spec.ref.name for Run %s/%s",
+			run.Namespace, run.Name)
+		return nil, nil, fmt.Errorf("missing spec.ref.name for Run %s", fmt.Sprintf("%s/%s", run.Namespace, run.Name))
+	}
+	return &CELMeta, &CELSpec, nil
```

Comment:
readability: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

e.g. you could rewrite this as:

```go
if ref == nil || name == "" {
  return nil, nil, err
}
cs, err := Get()
if err != nil {
  return nil, nil, err
}
return meta, spec, nil
```